warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UniqueExpressionTermsVisitorTest.java

```diff
@@ -0,0 +1,256 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.jexl.JexlASTHelper;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class UniqueExpressionTermsVisitorTest {
```

Comment:
Tests validating what we expect when hitting/exceeding the max iterations would be nice

Reply:
Testing proved that the max iterations is no longer needed. The original implementation required multiple iterations because of the preorder traversal. Now that traversal is post order the visitor only needs to visit the tree once.

Removed the max iteration code.